go-controller/pkg/node/routemanager/route_manager.go

```diff
@@ -89,7 +92,11 @@ func (c *Controller) Del(r netlink.Route) {
 	c.delRouteCh <- r
 }
 
-func (c *Controller) addRoute(r netlink.Route) error {
+// AddRoute attempts to add the route and returns with error
```

Comment:
@tssurya why not just use Add and not expose addRoute?

```
func (c *Controller) Add(r netlink.Route) {
	c.addRouteCh <- r
}
```

Reply:
let me see if I can change this, this specific change was done by Peri...

Reply:
@trozet : last time we had spoken to @martinkennelly he had mentioned its a pain to use blocking channels cause we can't process errors returned etc.. but for now I have incorporated your suggestion as a new commit, and if @pperiyasamy / @martinkennelly and you don't see concerns in that we can squash this.

Reply:
i remember we talked about this a good bit back - we wanted to be certain the routes were deleted before doing something else (which i cannot remember).

Reply:
i think route enqueue on the channel is not a problem, but dequeue happens on the another goroutine which asynchronous.  isn't it ? so this may cause vrf device delete happens first followed by delete routes associated with it.

Reply:
hmm not sure I follow the enqueue/dequeue problem.

Reply:
Surya and I looked through the code some more. It is definitely asynchronous which provides no feedback to the caller trying to add/del the route. Therefore if EgressIP or NetworkAdd add the route, but it cannot be added, those callers will not return error. The async loop will fail to add the route with an error logged, and then 2 minutes later retry. Therefore a user might think their network was added correctly, but in reality the route is missing.

The async model helps with ensuring threads are not blocked while trying to add routes, but it sacrifices feedback to the callers. I think it is a better idea to keep things synchronous for now until we identify thread contention as a real issue. This is kind of outside the scope of this PR, so I think for this PR @tssurya should just leverage the existing method to add the routes async.

I'll push another PR that refactors the controller to force it to be synchronous calls and we can discuss it in that PR.

Reply:
ack I am fine with that approach. thanks @trozet !